lib/fog/azurerm/storage.rb

```diff
@@ -3,10 +3,17 @@ module Fog
   module Storage
     # This class registers models, requests and collections
     class AzureRM < Fog::Service
-      requires :tenant_id
-      requires :client_id
-      requires :client_secret
-      requires :subscription_id
+      # Recognizes when creating management client
+      recognizes :tenant_id
+      recognizes :client_id
+      recognizes :client_secret
+      recognizes :subscription_id
+
+      # Recognizes when creating data client
+      recognizes :azure_storage_account_name
+      recognizes :azure_storage_access_key
+      recognizes :azure_storage_connection_string
+      recognizes :debug
```

Comment:
Agree, so I use "recognizes" instead of "requires" to register the class attributes. User can choose use the parameter set for storage accounts or storage data. They are not required in creating the Fog::Storage::AzureRM instance.
Both the following calls should work.
```ruby
# Creating an instance to operate on storage accounts
azure_storage_service = Fog::Storage.new(
  :provider => 'AzureRM',
  :tenant_id => '<Tenantid>',     
  :client_id =>    '<Clientid>', 
  :client_secret => '<ClientSecret>',
  :subscription_id => '<Subscriptionid>'
)
```
or
```ruby
# Creating an instance to operate on storage data
azure_storage_service = Fog::Storage.new(
  :provider => 'AzureRM',
  :azure_storage_account_name => '<account_name>',
  :azure_storage_access_key =>    '<account_key>'
)
```